### PR TITLE
Resolve ZAP testing CSP header errors using nonce on views

### PR DESCRIPTION
**Pull request by kaderhashmi**
Removing unsafe-inline from the CSP headers caused existing inline scripts (and styles) to throw errors in the browser. Adding the nonce to the views in conjunction with the existing NWebSec package resolved these issues

### CODE REVIEW

File: DFC.Composite.Shell.Views.UnitTests/Services.ViewRenderer/HtmlSupportTemplateBase.cs [hunk 1]
```diff
@@ -7,13 +9,20 @@ public class HtmlSupportTemplateBase<T> : TemplateBase<T>
         public HtmlSupportTemplateBase()
         {
             Html = new RazorHtmlHelper();
+            Component = A.Fake<IViewComponentHelper>();
+            //new DefaultViewComponentHelper(
+            //    new DefaultViewComponentDescriptorCollectionProvider(new DefaultViewComponentDescriptorProvider(new ApplicationPartManager())),
+            //    HtmlEncoder.Default,
+            //    new DefaultViewComponentSelector(new DefaultViewComponentDescriptorCollectionProvider(new DefaultViewComponentDescriptorProvider(new ApplicationPartManager()))),
+            //    A.Fake<IViewComponentInvokerFactory>(),
+            //    null);
```
Comment (Muthuramana) [inline DFC.Composite.Shell.Views.UnitTests/Services.ViewRenderer/HtmlSupportTemplateBase.cs]: Can this be removed?

File: DFC.Composite.Shell/Startup.cs [hunk 2]
```diff
@@ -80,11 +80,11 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
+            .StyleSources(s => s.Self().CustomSources($"{cdnLocation}/{Constants.NationalCareersToolkit}/css/"))
             .FontSources(s => s.Self().CustomSources($"{cdnLocation}/{Constants.NationalCareersToolkit}/fonts/"))
-            .ImageSources(s => s.Self().CustomSources(new string[] { $"{cdnLocation}/{Constants.NationalCareersToolkit}/images/", "www.google-analytics.com", "*.doubleclick.net" }))
-            .ConnectSources(s => s.Self().CustomSources($"{Configuration.GetValue<string>(Constants.ApplicationInsightsConnectSources)}")));
+            .ImageSources(s => s.Self().CustomSources($"{cdnLocation}/{Constants.NationalCareersToolkit}/images/", "www.google-analytics.com", "*.doubleclick.net"))
+            .ConnectSources(s => s.Self().CustomSources($"{Configuration.GetValue<string>(Constants.ApplicationInsightsConnectSources)}", "https://dc.services.visualstudio.com/")));
```
Comment (Muthuramana) [inline DFC.Composite.Shell/Startup.cs]: Can we format this 1 line per fluent func

File: DFC.Composite.Shell/Views/Shared/_ValidationScriptsPartial.cshtml [hunk 3]
```diff
@@ -1,18 +1,20 @@
 <environment include="Development">
-    <script src="~/lib/jquery-validation/jquery.validate.js"></script>
-    <script src="~/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.js"></script>
+    <script src="~/lib/jquery-validation/jquery.validate.js" nws-csp-add-nonce="true"></script>
+    <script src="~/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.js" nws-csp-add-nonce="true"></script>
```
Comment (Muthuramana) [inline DFC.Composite.Shell/Views/Shared/_ValidationScriptsPartial.cshtml]: Delete these lines, should be using dfc-digital-assets or CDN

File: DFC.Composite.Shell/Views/Shared/_ValidationScriptsPartial.cshtml [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+            nws-csp-add-nonce="true">
     </script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-validation-unobtrusive/3.2.11/jquery.validate.unobtrusive.min.js"
             asp-fallback-src="~/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.min.js"
             asp-fallback-test="window.jQuery && window.jQuery.validator && window.jQuery.validator.unobtrusive"
             crossorigin="anonymous"
-            integrity="sha256-9GycpJnliUjJDVDqP0UEu/bsm9U+3dnQUH8+3W10vkY=">
+            integrity="sha256-9GycpJnliUjJDVDqP0UEu/bsm9U+3dnQUH8+3W10vkY="
+            nws-csp-add-nonce="true">
     </script>
```
Comment (Muthuramana) [inline DFC.Composite.Shell/Views/Shared/_ValidationScriptsPartial.cshtml]: Should be using CDN, can we remove this pls?

File: DFC.Composite.Shell.Views.UnitTests/Services.ViewRenderer/HtmlSupportTemplateBase.cs (hunk 1 above)
Comment (kaderhashmi) [reply DFC.Composite.Shell.Views.UnitTests/Services.ViewRenderer/HtmlSupportTemplateBase.cs]: Redundant code cleared

File: DFC.Composite.Shell/Startup.cs (hunk 2 above)
Comment (kaderhashmi) [reply DFC.Composite.Shell/Startup.cs]: Amended as suggested

File: DFC.Composite.Shell/Views/Shared/_ValidationScriptsPartial.cshtml (hunk 3 above)
Comment (kaderhashmi) [reply DFC.Composite.Shell/Views/Shared/_ValidationScriptsPartial.cshtml]: Removed

File: DFC.Composite.Shell/Views/Shared/_ValidationScriptsPartial.cshtml (hunk 4 above)
Comment (kaderhashmi) [reply DFC.Composite.Shell/Views/Shared/_ValidationScriptsPartial.cshtml]: Removed

**Pull request merged**